src/products/components/ProductUpdatePage/formChannels.ts

```diff
@@ -40,18 +40,27 @@ export function useProductChannelListingsForm(
     touched.current = uniq([...touched.current, id]);
   };
 
-  const handleChannelChange = useCallback((id: string, data: ChannelOpts) => {
-    setChannels(prevData => ({
-      ...prevData,
-      updateChannels: prevData.updateChannels.map(prevListing =>
-        prevListing.channelId === id
-          ? { ...prevListing, ...data }
-          : prevListing,
-      ),
-    }));
-    triggerChange();
-    touch(id);
-  }, []);
+  const handleChannelChange = useCallback(
+    (id: string, data: ChannelOpts) => {
+      setChannels(prevData => ({
+        ...prevData,
+        updateChannels: prevData.updateChannels.map(prevListing =>
+          prevListing.channelId === id
+            ? {
+                ...prevListing,
+                ...data,
+                availableForPurchaseDate: data.isAvailableForPurchase
+                  ? data.availableForPurchase
+                  : null,
+              }
+            : prevListing,
+        ),
+      }));
+      triggerChange();
+      touch(id);
```

Comment:
could you somehow organize it in separate functions so it can be more readable?

Reply:
Done - please check 😊